Sources/Sentry/SentryTracer.m

```diff
@@ -581,6 +587,12 @@ - (void)finishInternal
         }
     }];
 
+    if (self.finishMustBeCalled && !self.wasFinishCalled) {
+        SENTRY_LOG_DEBUG(
+            @"Not capturing transaction because finish was not called before timing out.");
+        return;
+    }
```

Comment:
`m`: Wouldn't it be cleaner to move this right below ⬇️ , so we avoid extra work?
https://github.com/getsentry/sentry-cocoa/blob/b695b615f3b6fe1c5eacc4b298572524446b83dc/Sources/Sentry/SentryTracer.m#L524-L527

Reply:
We need to remove the transaction from the scope if that is the case.